src/main/java/org/sigmah/server/dao/impl/ProjectHibernateDAO.java

```diff
@@ -77,4 +79,70 @@
 		return query.getResultList();
 	}
 
+	@Override
+	public Project updateProjectTeamMembers(Project project, List<User> teamMembers, List<Profile> teamMemberProfiles, User modifier) {
+		project.setTeamMembers(teamMembers);
+		project.setTeamMemberProfiles(teamMemberProfiles);
+		return persist(project, modifier);
+	}
+
+	@Override
+	public Set<Integer> findProjectIdsByTeamMemberIdAndOrgUnitIds(Integer userId, Set<Integer> orgUnitIds) {
+		if (orgUnitIds.isEmpty()) {
+			return Collections.emptySet();
+		}
+
+		// Disable the ActivityInfo filter on Userdatabase.
+		DomainFilters.disableUserFilter(em());
+
+		TypedQuery<Integer> query = em().createQuery("SELECT pr.id " +
+			"FROM Project pr " +
+			"WHERE pr.manager.id = :userId " +
+			"OR pr.owner.id = :userId " +
+			"OR :userId IN ( " +
+			" SELECT tm.id " +
+			" FROM pr.teamMembers tm " +
+			") " +
+			"OR :userId IN ( " +
+			" SELECT oup.user.id " +
+			" FROM pr.teamMemberProfiles tmp, OrgUnitProfile oup " +
+			" JOIN oup.profiles oupp " +
+			" WHERE tmp.id = oupp.id " +
+			" AND oup.orgUnit.id IN (:orgUnitIds) " +
+			") ", Integer.class);
+		query.setParameter("userId", userId);
+		query.setParameter("orgUnitIds", orgUnitIds);
+
+		return new HashSet<>(query.getResultList());
+	}
+
+	@Override
+	public List<Project> findProjectByTeamMemberIdAndOrgUnitIds(Integer userId, Set<Integer> orgUnitIds) {
```

Comment:
could that be factored with the above method given that only the select clause is different?